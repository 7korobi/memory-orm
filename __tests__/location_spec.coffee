require "./models/index"
{ Query, Set } = require "../src/index"

geo = require '~/yaml/work_geo_orm.yml'
Set.work_location.reset geo

drillup = (item)->
  while item
    expect item
    .toMatchSnapshot()
    item = item.work_location

drilldown = (list, idx)->
  while list?.length
    expect list.pluck("idx")
    .toMatchSnapshot()
    list = list[idx].work_locations.list

describe "work location", ->
  test 'query size', ->
    expect Query.work_locations.list.length
    .toMatchSnapshot()
    return

  test 'query size with zip', ->
    drillup Query.work_locations.sort("id_ary.length","desc").list[0]
    return

  test 'query size no zip', ->
    drillup Query.work_locations.where((o)-> ! o.zipcode ).sort("id_ary.length","desc").list[0]
    return

  test 'query for toplevel', ->
    drilldown Query.work_locations.where((o)-> ! o.work_location_id ).sort("id_ary.length","desc").list, 0
    return


  return

