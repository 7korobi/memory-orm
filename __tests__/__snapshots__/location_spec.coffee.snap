// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`work location query for toplevel 1`] = `
list [
  "北海道",
  "富山県",
  "新潟県",
  "神奈川県",
  "東京都",
  "千葉県",
  "埼玉県",
  "愛知県",
  "石川県",
  "福井県",
  "静岡県",
  "岐阜県",
  "山梨県",
  "群馬県",
  "栃木県",
  "茨城県",
  "長野県",
  "三重県",
  "滋賀県",
  "京都府",
  "青森県",
  "大阪府",
  "兵庫県",
  "福島県",
  "奈良県",
  "和歌山県",
  "岩手県",
  "鳥取県",
  "島根県",
  "岡山県",
  "広島県",
  "山口県",
  "徳島県",
  "山形県",
  "宮城県",
  "沖縄県",
  "鹿児島県",
  "宮崎県",
  "大分県",
  "熊本県",
  "長崎県",
  "佐賀県",
  "福岡県",
  "高知県",
  "愛媛県",
  "香川県",
  "秋田県",
]
`;

exports[`work location query for toplevel 2`] = `
list [
  "札幌市",
  "函館市",
  "小樽市",
  "旭川市",
  "室蘭市",
  "釧路市",
  "帯広市",
  "北見市",
  "夕張市",
  "岩見沢市",
  "網走市",
  "留萌市",
  "苫小牧市",
  "稚内市",
  "美唄市",
  "芦別市",
  "江別市",
  "赤平市",
  "紋別市",
  "士別市",
  "名寄市",
  "三笠市",
  "根室市",
  "千歳市",
  "滝川市",
  "砂川市",
  "歌志内市",
  "深川市",
  "富良野市",
  "登別市",
  "恵庭市",
  "伊達市",
  "北広島市",
  "石狩市",
  "北斗市",
  "石狩郡",
  "松前郡",
  "上磯郡",
  "亀田郡",
  "茅部郡",
  "二海郡",
  "山越郡",
  "檜山郡",
  "爾志郡",
  "奥尻郡",
  "瀬棚郡",
  "久遠郡",
  "島牧郡",
  "寿都郡",
  "磯谷郡",
  "虻田郡",
  "岩内郡",
  "古宇郡",
  "積丹郡",
  "古平郡",
  "余市郡",
  "空知郡",
  "夕張郡",
  "樺戸郡",
  "雨竜郡",
  "上川郡",
  "勇払郡",
  "中川郡",
  "増毛郡",
  "留萌郡",
  "苫前郡",
  "天塩郡",
  "宗谷郡",
  "枝幸郡",
  "礼文郡",
  "利尻郡",
  "網走郡",
  "斜里郡",
  "常呂郡",
  "紋別郡",
  "有珠郡",
  "白老郡",
  "沙流郡",
  "新冠郡",
  "浦河郡",
  "様似郡",
  "幌泉郡",
  "日高郡",
  "河東郡",
  "河西郡",
  "広尾郡",
  "足寄郡",
  "十勝郡",
  "釧路郡",
  "厚岸郡",
  "川上郡",
  "阿寒郡",
  "白糠郡",
  "野付郡",
  "標津郡",
  "目梨郡",
  "道庁",
  "千歳市都",
]
`;

exports[`work location query for toplevel 3`] = `
list [
  "中央区",
  "北区",
  "東区",
  "白石区",
  "豊平区",
  "南区",
  "西区",
  "厚別区",
  "手稲区",
  "清田区",
  "市役所",
]
`;

exports[`work location query for toplevel 4`] = `
list [
  "大通",
  "北一条",
  "北二条",
  "北三条",
  "北四条",
  "北五条",
  "北六条",
  "北七条",
  "北八条",
  "北九条",
  "北十条",
  "北十一条",
  "北十二条",
  "北十三条",
  "北十四条",
  "北十五条",
  "北十六条",
  "北十七条",
  "北十八条",
  "北二十条",
  "北二十一条",
  "北二十二条",
  "中島",
  "南一条",
  "南二条",
  "南三条",
  "南四条",
  "南五条",
  "南六条",
  "南七条",
  "南八条",
  "南九条",
  "南十条",
  "南十一条",
  "南十二条",
  "南十三条",
  "南十四条",
  "南十五条",
  "南十六条",
  "南十七条",
  "南十八条",
  "南十九条",
  "南二十条",
  "南二十一条",
  "南二十二条",
  "南二十三条",
  "南二十四条",
  "南二十五条",
  "南二十六条",
  "南二十七条",
  "南二十八条",
  "南二十九条",
  "南三十条",
  "宮の森四条",
  "旭ケ丘",
  "伏見",
  "界川",
  "円山西町",
  "盤渓",
  "双子山",
  "宮の森一条",
  "宮の森二条",
  "宮の森三条",
  "宮の森",
  "宮ケ丘",
]
`;

exports[`work location query for toplevel 5`] = `
list [
  "西",
  "東",
]
`;

exports[`work location query for toplevel 6`] = `
list [
  "（１〜１９丁目）",
  "（２０〜２８丁目）",
]
`;

exports[`work location query size 1`] = `131732`;

exports[`work location query size no zip 1`] = `
model {
  "_id": "大阪府-大阪市-北区-大淀中-梅田-スカイビルタワー-イースト",
  "idx": "イースト",
  "label": "イースト",
  "name": "大阪府大阪市北区大淀中梅田スカイビルタワーイースト",
  "unit": null,
  "work_location_id": "大阪府-大阪市-北区-大淀中-梅田-スカイビルタワー",
}
`;

exports[`work location query size no zip 2`] = `
model {
  "_id": "大阪府-大阪市-北区-大淀中-梅田-スカイビルタワー",
  "idx": "スカイビルタワー",
  "label": "スカイビルタワー",
  "name": "大阪府大阪市北区大淀中梅田スカイビルタワー",
  "unit": null,
  "work_location_id": "大阪府-大阪市-北区-大淀中-梅田",
}
`;

exports[`work location query size no zip 3`] = `
model {
  "_id": "大阪府-大阪市-北区-大淀中-梅田",
  "idx": "梅田",
  "label": "梅田",
  "name": "大阪府大阪市北区大淀中梅田",
  "unit": null,
  "work_location_id": "大阪府-大阪市-北区-大淀中",
}
`;

exports[`work location query size no zip 4`] = `
model {
  "_id": "大阪府-大阪市-北区-大淀中",
  "idx": "大淀中",
  "label": "大淀中",
  "name": "大阪府大阪市北区大淀中",
  "unit": null,
  "work_location_id": "大阪府-大阪市-北区",
}
`;

exports[`work location query size no zip 5`] = `
model {
  "_id": "大阪府-大阪市-北区",
  "idx": "北区",
  "label": "北",
  "name": "大阪府大阪市北区",
  "unit": "区",
  "work_location_id": "大阪府-大阪市",
}
`;

exports[`work location query size no zip 6`] = `
model {
  "_id": "大阪府-大阪市",
  "idx": "大阪市",
  "label": "大阪",
  "name": "大阪府大阪市",
  "unit": "市",
  "work_location_id": "大阪府",
}
`;

exports[`work location query size no zip 7`] = `
model {
  "_id": "大阪府",
  "idx": "大阪府",
  "label": "大阪",
  "name": "大阪府",
  "unit": "府",
}
`;

exports[`work location query size with zip 1`] = `
model {
  "_id": "大阪府-大阪市-北区-大淀中-梅田-スカイビルタワー-イースト-（１階）",
  "idx": "（１階）",
  "jiscode": "27127",
  "label": "（１階）",
  "name": "大阪府大阪市北区大淀中梅田スカイビルタワーイースト（１階）",
  "ruby": "おおさかふおおさかしきたくおおよどなかうめだすかいびるたわーいーすと1かい",
  "unit": null,
  "work_location_id": "大阪府-大阪市-北区-大淀中-梅田-スカイビルタワー-イースト",
  "zipcode": "5316001",
}
`;

exports[`work location query size with zip 2`] = `
model {
  "_id": "大阪府-大阪市-北区-大淀中-梅田-スカイビルタワー-イースト",
  "idx": "イースト",
  "label": "イースト",
  "name": "大阪府大阪市北区大淀中梅田スカイビルタワーイースト",
  "unit": null,
  "work_location_id": "大阪府-大阪市-北区-大淀中-梅田-スカイビルタワー",
}
`;

exports[`work location query size with zip 3`] = `
model {
  "_id": "大阪府-大阪市-北区-大淀中-梅田-スカイビルタワー",
  "idx": "スカイビルタワー",
  "label": "スカイビルタワー",
  "name": "大阪府大阪市北区大淀中梅田スカイビルタワー",
  "unit": null,
  "work_location_id": "大阪府-大阪市-北区-大淀中-梅田",
}
`;

exports[`work location query size with zip 4`] = `
model {
  "_id": "大阪府-大阪市-北区-大淀中-梅田",
  "idx": "梅田",
  "label": "梅田",
  "name": "大阪府大阪市北区大淀中梅田",
  "unit": null,
  "work_location_id": "大阪府-大阪市-北区-大淀中",
}
`;

exports[`work location query size with zip 5`] = `
model {
  "_id": "大阪府-大阪市-北区-大淀中",
  "idx": "大淀中",
  "label": "大淀中",
  "name": "大阪府大阪市北区大淀中",
  "unit": null,
  "work_location_id": "大阪府-大阪市-北区",
}
`;

exports[`work location query size with zip 6`] = `
model {
  "_id": "大阪府-大阪市-北区",
  "idx": "北区",
  "label": "北",
  "name": "大阪府大阪市北区",
  "unit": "区",
  "work_location_id": "大阪府-大阪市",
}
`;

exports[`work location query size with zip 7`] = `
model {
  "_id": "大阪府-大阪市",
  "idx": "大阪市",
  "label": "大阪",
  "name": "大阪府大阪市",
  "unit": "市",
  "work_location_id": "大阪府",
}
`;

exports[`work location query size with zip 8`] = `
model {
  "_id": "大阪府",
  "idx": "大阪府",
  "label": "大阪",
  "name": "大阪府",
  "unit": "府",
}
`;
